src/network/networkd-link.c

```diff
@@ -457,6 +457,18 @@ static int link_new(Manager *manager, sd_netlink_message *message, Link **ret) {
         if (r < 0)
                 return r;
 
+        r = hashmap_ensure_allocated(&manager->links_by_name, &string_hash_ops);
+        if (r < 0)
+                return r;
+
+        r = hashmap_put(manager->links_by_name, link->ifname, link);
+        if (r < 0)
+                return r;
```

Comment:
maybe move all this also into link_update_by_name(), so that that function comprehensively adds all names to the hashmap both the main one and the alternative ones, and implicitly creates the hashmap before, if it is missing